Core/Core/SwiftUIViews/TopBar/ViewModel/TopBarViewModel.swift

```diff
@@ -19,7 +19,7 @@
 import SwiftUI
 
 public class TopBarViewModel: ObservableObject {
-    public let items: [TopBarItemViewModel]
+    public var items: [TopBarItemViewModel]
```

Comment:
This property isn't `@Published` so changing it won't trigger the necessary SwiftUI updates. Also, the `selectedItemIndex` could become invalid if I set a smaller `items` array than it was and the selection was on the last element. Changing this array also raises the question what to do with the selection if a completely different array is set. If we keep the selection index at the same position but the underlying TopBarItemViewModel changes it changes the active menu on the UI.
I know most of these don't affect the current behaviour but since this is a shared component we should make sure that we can't break it easily nor it requires "hidden" knowledge. (Like you should manually update the selectedItemIndex property.)
An easy fix would be to revert this change and move the initialization of the `topBarViewModel` in `K5DashboardCardViewModel` to the init method.